### PR TITLE
UI: Enable high DPI scaling, for Qt >= 5.6

### PR DESCRIPTION
**Pull request by k-ross**

### CODE REVIEW

**Comment by k-ross** [comment]
I tried following an online guide to squashing, but it just messed things up. Deleted the branch and pull request, and started over.

**Comment by dodgepong** [comment]
This used to be in the program (https://github.com/obsproject/obs-studio/pull/650) but it was reverted (https://github.com/obsproject/obs-studio/commit/06894c6e84d1a12bce4cbfd635cdb17065f1e496, https://github.com/obsproject/obs-studio/commit/fa0bd585a56463b61de11cdddedc427e822e3c80) due to some unexpected behavior that it caused with things like weird title bar heights and not handling monitors with different DPIs well. It's possible things have been improved since then but we'll want to test this pretty thoroughly.

**Comment by k-ross** [comment]
Are there any bug reports associated with those reverts? I'd be willing to re-test, if I had some specifics.

**Comment by dodgepong** [comment]
Unfortunately I don't think there were any formal bug reports, just an observation and confirmation on IRC. There may be others who can be more specific? @DDRBoxman

**Comment by DDRBoxman** [comment]
There are weird things like moving between monitors with different DPIs where the titlebar will be the wrong size. I honestly don't remember the other issues, the title bar mostly stood out. But a multi DPI environment on windows will suss most of them out.

**Comment by SuslikV** [comment]
You need to test it on W7 too because of default DPI Awareness Level for Qt application. (if anybody uses W7)

File: UI/obs-app.cpp [hunk 1]
```diff
@@ -1645,7 +1645,9 @@ static int run_program(fstream &logFile, int argc, char *argv[])
 	profile_register_root(run_program_init, 0);
 
 	ScopeProfiler prof{run_program_init};
-
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 6, 0))
```
[the remark this answers is not in the document]
Comment (SuslikV) [reply UI/obs-app.cpp]: @DDRBoxman , It is not needed. Or you wish to solve all Qt bugs manually? Option introduced in Qt 5.6.0.

Qt 5.11.2 (and 5.12.0) for example has stylesheet bug with checked menus (when customized). Do you wish to fix it too with the custom painting for example?

**Comment by DDRBoxman** [comment]
Changing this in so that it will enable on qt 5.11 since older versions have some DPI related bugs. Next release will most likely not have this enabled yet as there are a few issues to fix. But once we use 5.11 or newer it should be good to go!

Look for a5fac7d78cf1193f7d07b14ad8f662b74e76937f

**Comment by SuslikV** [comment]
Was mentioned in:
https://obsproject.com/mantis/view.php?id=1358

**Pull request closed without merging**